enhancements/network/ingress-node/ingress-node-firewall-with-ebpf.md

```diff
@@ -0,0 +1,465 @@
+	// +optional
+	ProtocolRule IngressNodeFirewallProtoRule `json:"protoRule"`
+
+	// IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols.
+	// +optional
+	ICMPRule IngressNodeFirewallICMPRule `json:"icmpRule"`
+
+	// Protocol can be ICMP, ICMPv6, TCP, SCTP or UDP.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="icmp";"icmpv6";"tcp";"udp";"sctp"
+	// +optional
+	Protocol IngressNodeFirewallRuleProtocolType `json:"protocol"`
+
+	// Action can be allow or deny.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="allow";"deny"
+	// +optional
+	Action IngressNodeFirewallActionType `json:"action"`
+}
+
+// ProtocolType defines the protocol types that are supported
+type IngressNodeFirewallRuleProtocolType string
+
+const (
+	// ProtocolTypeICMP refers to the ICMP protocol
+	ProtocolTypeICMP IngressNodeFirewallRuleProtocolType = "icmp"
+
+	// ProtocolTypeICMPv6 refers to the ICMPv6 protocol
+	ProtocolTypeICMPv6 IngressNodeFirewallRuleProtocolType = "icmpv6"
+
+	// ProtocolTypeTCP refers to the TCP protocol, for either IPv4 or IPv6
+	ProtocolTypeTCP IngressNodeFirewallRuleProtocolType = "tcp"
+
+	// ProtocolTypeUDP refers to the UDP protocol, for either IPv4 or IPv6
+	ProtocolTypeUDP IngressNodeFirewallRuleProtocolType = "udp"
+
+	// ProtocolTypeSCTP refers to the SCTP protocol, for either IPv4 or IPv6
+	ProtocolTypeSCTP IngressNodeFirewallRuleProtocolType = "sctp"
+)
+
+// IngressNodeFirewallActionType indicates whether an IngressNodeFirewallRule allows or denies traffic
+// +kubebuilder:validation:Pattern=`^allow|deny$`
+type IngressNodeFirewallActionType string
+
+const (
+	IngressNodeFirewallAllow IngressNodeFirewallActionType = "allow"
+	IngressNodeFirewallDeny  IngressNodeFirewallActionType = "deny"
+)
+
+// IngressNodeFirewallRules define ingress node firewall rule
+type IngressNodeFirewallRules struct {
+	// FromCIDRS is A list of CIDR from which we apply node firewall rule
+	// +kubebuilder:validation:one-of[]=Type=string;Format:cidr
+	FromCIDRs []string `json:"fromCIDRs"`
+	// FirewallProtocolRules is A list of per protocol ingress node firewall rules
+	FirewallProtocolRules []IngressNodeFirewallProtocolRule `json:"rules"`
+}
+
+// IngressNodeFirewallSpec defines the desired state of IngressNodeFirewall
+type IngressNodeFirewallSpec struct {
+	// Selects node(s) where ingress firewall rules will be applied to
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// A list of ingress firewall policy rules.
+	// empty list indicates no ingress firewall i.e allow all incoming traffic.
+	// +kubebuilder:validation:Optional
+	// +optional
+	Ingress []IngressNodeFirewallRules `json:"ingress"`
+
+	// A list of interfaces where the ingress firewall policy will be applied on.
+	// empty list indicates the firewall policy applied on all interfaces
```

Comment:
What is meant by all interfaces? Veths included? I don't think that is what we want.